IPIP/0379-delegated-ipns-http-api.md

```diff
@@ -0,0 +1,82 @@
+The performance of naming system over the IPFS DHT can suffer from long delays due to churn of records and quorum requirements.
+
+## HTTP API Specification
+
+See [Delegated IPNS HTTP API](../routing/DELEGATED_IPNS_HTTP.md) specification.
+
+## Design rationale
+
+The rationale for delegated IPNS over HTTP APIs closely follows the reasoning listed in [IPIP-337](0337-delegated-routing-http-api.md#design-rationale).
+
+The document proposes the following:
+- Use of HTTP semantics for publication and resolution of naming records.
+- Preference for human-readable request and response encoding, such as JSON format
+- Optional backward compatibility support for the existing ProtocolBuffer format using `Content-Type: application/vnd.ipfs.ipns-record`
+- Use of extra headers in `OPTIONS` response to communicate the supported capabilities and limitations, namely:
+  - `X-Ipns-Allow-Max-Size` -- to signal maximum supported IPNS record size
+  - `X-Ipns-Allow-Protobuf` -- to signal whether the server supports ProtocolBuffer formatted records.
+- Streaming interaction is not supported.
+
+### User benefit
+
+The ability of offload naming onto another process or server via an idiomatic and simple HTTP API opens up an opportunity to not only reduce the burden of naming system management, but also introduce alternative naming systems across other routing systems, namely IPNI.
+
+### Compatibility
+
+#### Backwards Compatibility
+
+##### Serialization Format
+
+IPNS records use [ProtocolBuffer serialisation format](../ipns/IPNS.md#record-serialization-format).
+This format is widely in use in IPNS over PubSub and DHT routing systems.
+One of the motivations of this document is to introduce simple to use HTTP APIs and ultimately reduce barrier for interaction across alternative systems.
+Further, interoperability across the existing and HTTP APIs is also desirable in order to reduce the barrier for adoption of the delegated HTTP APIs.
+
+The specification here maintains backwards compatibility in terms of record serialisation, with preference for human-readable formats such as JSON.
+To maximize interoperability with existing ecosystem, the canonical [IPNS record serialization format](../ipns/IPNS.md#record-serialization-format) (`0x0300`) can be requested with content type `application/vnd.ipfs.ipns-record`.
+
+##### Reframe
+
+See [IPIP-337/Backwards Compatibility](0337-delegated-routing-http-api.md#backwards-compatibility).
+
+#### Forwards Compatibility
+
+See [IPIP-337/Forwards Compatibility](0337-delegated-routing-http-api.md#forwards-compatibility).
+
+### Security
+
+All interaction over the APIs should use TLS to protect against third-party observation and tampering.
+Additionally, the IPNS records are signed by the publisher's identity and contain sequence number to avoid replay attacks.
+
+To avoid Denial of Service attack, maximum IPNS record size of `10 KiB` applies.
```

Comment:
```suggestion
To avoid Denial of Service attack, [maximum IPNS record size defined in IPNS spec](https://specs.ipfs.tech/ipns/ipns-record/#record-size-limit) applies.
```